tools/marvin/marvin/lib/utils.py

```diff
@@ -520,4 +520,22 @@ def verifyRouterState(apiclient, routerid, allowedstates):
     if routers[0].state.lower() not in allowedstates:
         return [FAIL, "state of the router should be in %s but is %s" %
             (allowedstates, routers[0].state)]
-    return [PASS, None]
\ No newline at end of file
+    return [PASS, None]
+
+
+
+def wait_until(retry_interval=2, no_of_times=2, callback=None, *callback_args):
+    """ Utility method to try out the callback method at most no_of_times with a interval of retry_interval,
+    Will return immediately if callback returns True. The callback method should be written to return a list of values first being a boolean """
+
+    if callback is None:
+        return INVALID_INPUT
```

Comment:
Due to following two reason I preferred returning an error code instead of an exception.
1. Returning a pre-defined error code makes it  usage more flexible as raising an exception or continue will be defined by the user of the method.
2. "INVALID_INPUT" is a Marvin error code used by other utility methods to signal bad inputs and this maintains that pattern.